### DISCUSSION TITLE
Converting to WFDB format

### DISCUSSION THREAD

**Discussion by briangow** [chorus-ai/chorus_waveform#89] [General]
Tools for converting the waveforms for CHoRUS into WFDB format need to be provided. We will use this discussion to share ideas on how this should be accomplished.

**Comment by briangow** [comment]
We need to decide whether the conversion to WFDB will be handled at the site level or at CHoRUS central.

**Reply by del42** [reply]
I would say both. Sites that directly work with raw signals are more familiar with their signal nuances. However, there are sites that simply request and deliver, which would require conversion on CHoRUS Central

**Comment by briangow** [comment]
We need to come up with a plan that considers all of the sites needs for getting their waveforms into WFDB format. 

If we plan to convert to WFDB at the site level, it may be ideal to try to go back to the monitor "format" and convert from that. I know that there are at least some sites working with these monitor formats:

- Philips DWC
  - [Downcast](https://github.com/MIT-LCP/downcast), [PreVent](https://github.com/Ostrich-Emulators/PreVent?tab=readme-ov-file#command-line-options), direct conversion from latest DWC software 
- GE Bedmaster as STPGE
  - PreVent
- GE Capsule as HL7 feed

I've added tools that I'm aware of which may help with converting from these monitor formats as sub-bullets. While PreVent claims it converts from DWC and STPGE to WFDB, my understanding is that exporting in WFDB is not currently working as expected and would require additional development. 

If we don't have access to the monitor format at some sites, we may need to consider converting from other formats. Some sites are currently using Parquet as the format they are submitting their waveforms in. If we need to handle converting from formats like Parquet, which don't have a physiological waveform specification, we may need to design converters that are site-specific.

**Reply by del42** [reply]
Sickbay? I think they are HL7 feed.
